### PR TITLE
Add ClusterSingleton supervision documentation

### PR DESCRIPTION
**Pull request by JustinPihony**

### CODE REVIEW

File: akka-docs/src/main/paradox/cluster-singleton.md [hunk 1]
```diff
@@ -156,4 +156,75 @@ or create it from another config section with the same layout as below. `Cluster
...
\ No newline at end of file
+@@snip [reference.conf]($akka$/akka-cluster-tools/src/main/resources/reference.conf) { #singleton-proxy-config }
+
+## Supervision
+
+Sometimes it is useful to add supervision for the Cluster Singleton itself. To accomplish this you need to add a parent supervisor actor which will be used to create the 'real' singleton instance. Below is an example implementation (credit to [this StackOverflow answer](https://stackoverflow.com/a/36716708/779513))
+
+Scala
+:  ```
+import akka.actor.{ Actor, Props }
```
Comment (chbatey) [inline akka-docs/src/main/paradox/cluster-singleton.md]: Can we put this code into a test? Otherwise it'll end up not compiling

**Comment by chbatey** [comment]
OK TO TEST

**Comment by chbatey** [comment]
Ping @JustinPihony :)

**Comment by JustinPihony** [comment]
@chbatey Thanks for the ping - this is now ready with compilable code

File: akka-docs/src/main/paradox/cluster-singleton.md [hunk 2]
```diff
@@ -156,4 +156,23 @@ or create it from another config section with the same layout as below. `Cluster
...
+:  @@snip [SupervisorActor.java]($akka$/akka-docs/src/test/java/jdocs/cluster/singleton/SupervisorActor.java) { #singleton-supervisor-actor }
+
+And used here
+
+Scala
+:  @@snip [ClusterSingletonSupervision.scala]($akka$/akka-docs/src/test/scala/docs/cluster/singleton/ClusterSingletonSupervision.scala) { #singleton-supervisor-actor-usage }
+
+Java
+:  @@snip [ClusterSingletonSupervision.java]($akka$/akka-docs/src/test/java/jdocs/cluster/singleton/ClusterSingletonSupervision.java) { #singleton-supervisor-actor-usage-imports }
+@@snip [ClusterSingletonSupervision.java]($akka$/akka-docs/src/test/java/jdocs/cluster/singleton/ClusterSingletonSupervision.java) { #singleton-supervisor-actor-usage }
```
Comment (ktoso) [inline akka-docs/src/main/paradox/cluster-singleton.md]: just sanity checking if this renders okey?

Comment (JustinPihony) [reply akka-docs/src/main/paradox/cluster-singleton.md]: It seemed to for me, but I'll admit that the full styling wasn't working so I can't be 100%

Comment (raboof) [reply akka-docs/src/main/paradox/cluster-singleton.md]: I think I've seen this sometimes work and sometimes not - IIRC a safer way is to put it on one line like `@@snip [ClusterSingletonSupervision.java]($akka$/akka-docs/src/test/java/jdocs/cluster/singleton/ClusterSingletonSupervision.java) { #singleton-supervisor-actor-usage-imports #singleton-supervisor-actor-usage }`

Comment (JustinPihony) [reply akka-docs/src/main/paradox/cluster-singleton.md]: Thanks @raboof   I made that adjustment :)

Comment (ktoso) [reply akka-docs/src/main/paradox/cluster-singleton.md]: This was fine actually; it looked quite good; now the code is not valid Java code and looks weird; you can not import like this in-line in Java. @raboof

Comment (raboof) [reply akka-docs/src/main/paradox/cluster-singleton.md]: Oops sorry for the distraction then!

Comment (ktoso) [reply akka-docs/src/main/paradox/cluster-singleton.md]: This looked good and more "real":
cluster_singleton_ _akka_documentation

Comment (ktoso) [reply akka-docs/src/main/paradox/cluster-singleton.md]: No worries, I guess I'll revert the change and merge...

Comment (raboof) [reply akka-docs/src/main/paradox/cluster-singleton.md]: jup if that works that seems neat

Comment (JustinPihony) [reply akka-docs/src/main/paradox/cluster-singleton.md]: I just reverted manually

Comment (ktoso) [reply akka-docs/src/main/paradox/cluster-singleton.md]: thanks! :)

**Review by ktoso — APPROVED** [review]

**Pull request merged**